qiskit_metal/toolbox_metal/bounds_for_path_and_poly_tables.py

```diff
@@ -0,0 +1,234 @@
+# -*- coding: utf-8 -*-
+from typing import List, Tuple, Union
+
+from qiskit_metal.designs import MultiPlanar
+from qiskit_metal.toolbox_python.utility_functions import determine_larger_box
+
+import pandas as pd
+
+
+class BoundsForPathAndPolyTables():
+    """Create class which can be used by multiple renderers.  In particular, this class
+    assumes a LayerStack is being used within QDesign.
+    """
+
+    def __init__(self, design: MultiPlanar):
+        self.design = design
+        self.chip_names_matched = None  # bool
+        self.valid_chip_names = None  # set of valid chip names from layer_stack
+
+    def get_bounds_of_path_and_poly_tables(
+        self, box_plus_buffer: bool, qcomp_ids: List, case: int, x_buff: float,
+        y_buff: float
+    ) -> tuple[tuple[float, float, float, float], pd.DataFrame, bool, Union[
+            None, set]]:
+        """If box_plus_buffer is True, returns a tuple containing minx, miny, maxx, maxy values for the
+        bounds of what was selected to be rendered.
+        Else, use the chip size for Tuple.
+
+        Also, return the Tuple within the class to can be used later within a renderer.
+
+        Args:
+            box_plus_buffer (bool): Use the box of selected components plus a buffer size OR
+                                the chip size based on QComponents selected.
+            qcomp_ids (List):  Empty or partial list of components in QDesign. From QRenderer.get_unique_component_ids
+            case (int): Return code used from QRenderer.get_unique_component_ids()
+            x_buff (float): If box_plus_buffer, need the buffer value in x coordinate.
+            y_buff (float): If box_plus_buffer, need the buffer value in y coordinate.
+
+        Returns:
+            tuple[tuple[float, float, float, float], pd.DataFrame, bool, Union[None,set]]:
+                                    tuple: minx, miny, maxx, maxy values based
+                                                    on either total chip size or box_plus_buffer.
+                                    pd.DataFrame: The path and poly dataframes concatenated for qcomp_ids
+                                    bool: If there is a key in design with same chip_name as in layer_stack
+                                    Union[None,set]: Either None if the names don't match,
+                                                    or the set of chip_names that can be used.
+        """
+        box_for_ansys = None
```

Comment:
Maybe consider refactoring this to `stack_xy_bounds` instead of `box_for_ansys`?

Reply:
I think you are looking for bounds per layer.  However, I wanted bounds for all of the rendering.

Reply:
name changed and pushed